service/keys/keys.go

```diff
@@ -12,6 +16,8 @@ type Keys interface {
 	GetNostrPublicKey() string
 	// Wallet Service Nostr secret key
 	GetNostrSecretKey() string
+	// GetBIP32ChildKey derives a BIP32 child key from the nostrSecretKey given a child key index
```

Comment:
Should we add TODOs to above to remove the old method after a year so we can remove conditionals (and delete all apps that don't have a wallet pubkey)?

Reply:
How do we make sure that people upgrading will not have any legacy apps anymore?

Reply:
Yeah, it seems like it might be best to rethink this and store encrypted private keys in the app table. Then we don't need any conditional logic.

Reply:
One issue I see with saving priv keys on DB for the legacy apps is that since all legacy apps will have to use the same (existing single) priv/pub wallet key (so not to break existing connections with clients), we still need conditional code to handle not deleting the single nip47 info event (kind 13194) from the relay when legacy app gets deleted, until it is the last legacy app in hub, right?

Reply:
I guess so, but only this one simple check. It's a big improvement over the backward compatibility issues we would have with not saving the encrypted private keys in the app table.

Reply:
The backward compatibility needed are only 2 simple checks (if WalletPubKey column is empty will use the single master key). Saving the encrypted private keys seems to me unnecessary (only for the sake of eliminating 2 simple checks) and more complicated to maintain.